### PR TITLE
feat: support ANSI color indices in theme configuration

### LINKED ISSUE #770: Support ANSI color indices in theme configuration
Hi! First off, thanks for gh-dash — it's become a daily driver for me and it's a pleasure to use.

I'd love to be able to reference ANSI color indices (0–15) in the theme config, in addition to hex values. The motivation is pretty simple: I use a terminal color scheme (Solarized) and I regularly switch between dark and light variants depending on ambient light. Since my 16 base ANSI colors are remapped by the terminal theme, apps that reference ANSI indices automatically look right in both modes — no config change needed.

Right now, the `HexColor` type enforces hex-only values (`#RRGGBB` / `#RGB`), so there's no way to tap into the terminal palette. I end up having to hardcode hex values that only match one of my two themes.

I took a quick look at the code and it seems like lipgloss already supports this natively — `lipgloss.Color("1")` resolves to ANSI color 1, and `CompleteColor` / `CompleteAdaptiveColor` allow specifying values per color profile. So the change might mostly be about relaxing the config schema to accept ANSI indices alongside hex, and passing them through to lipgloss as-is.

Concretely, something like this in `config.yml` would be amazing:

```yaml
theme:
  colors:
    text:
      primary: "12"      # ANSI bright blue
      secondary: "10"    # ANSI bright green
      faint: "11"        # ANSI bright yellow
    border:
      primary: "4"       # ANSI blue
```

I'm happy to take a stab at a PR if you think this is worth pursuing and can point me in the right direction.

Thanks again for all your work on this!

### PR DESCRIPTION
**Pull request by graelo**
Hello again, thanks for your kind message in #770.

Here's my PR. I tested it by
- running tests (added 3 new tests)
- building locally with `go build -o gh-dash .` and moving the binary to `~/.local/share/gh/extensions/gh-dash/` and then running `gh dash`, with the config file below.

In a nutshell, this PR replaces the `hexcolor` validator with a custom `color` validator that accepts both hex and ansi (0-255) formats. It's a strictly additive change — all previously valid configs remain valid.

Config file and screenshot:
<details>
`~/.config/gh-dash/config.yml`

```yml
theme:
  ui:
    sectionsShowCount: true
    table:
      showSeparators: true
      compact: false
  colors:
    text:
      primary: '2' # green
      secondary: '6' # cyan
      inverted: '15' # base3 (brightwhite)
      faint: '8' # base03 (brightblack)
      warning: '1' # red
      success: '2' # green
      actor: '4' # blue
    background:
      selected: '7' # base2 (white)
    border:
      primary: '0' # black
      secondary: '5' # magenta
      faint: '14' # base1 (brightcyan)
    icon:
      newcontributor: '037'
      contributor: '133'
      collaborator: '136'
      member: '136'
      owner: '166'
      unknownrole: '125'
  icons:
    newcontributor: "\U000F0394"
    contributor: ''
    collaborator: ''
    member: ''
    owner: ''
    unknownrole: "\U000F0B59"
```

Screenshot 2026-02-12 at 21 48 30

</details>

### Changes

- Renamed the `HexColor` type to `Color` and swapped the built-in `hexcolor` tag for a custom `color` validator that accepts hex colors or numeric ANSI indices (0–255)
- Renamed the `_shimHex` helper in `theme.go` to `_shimColor` (pure rename, no logic change — lipgloss already handles both formats)
- Updated the documentation schemas to reflect the broader accepted format
- Added tests for valid and invalid inputs, including edge cases (`"256"`, `"-1"`, leading zeros, etc.)

Closes #770

### CODE REVIEW

**Comment by graelo** [comment]
I just saw the lint issue, on it!

**Comment by graelo** [comment]
@dlvhdr Hi, I think there's a CI issue with the docker workflow. It looks unrelated to my PR.

**Comment by dlvhdr** [comment]
Yeah it's unrelated. Still need to review the code 🙏

File: docs/src/data/schemas/theme.yaml [hunk 1]
```diff
@@ -226,12 +226,12 @@ properties:
...
             type: string
             default: "#c6c6c6"
-            pattern: ^#([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$
+            pattern: ^(#([a-fA-F0-9]{6}|[a-fA-F0-9]{3})|([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))$
           inverted:
             title: Inverted Text Color
             description: >-
-              Specifies the color for text on an inverted background. Must be a valid hex color,
-              like `#a3c` or `#aa33cc`.
+              Specifies the color for text on an inverted background. Must be a valid hex color
```
Comment (dlvhdr) [inline docs/src/data/schemas/theme.yaml:233]: we also need to update `docs/src/content/docs/configuration/theme.mdx`

Comment (graelo) [reply docs/src/data/schemas/theme.yaml:233]: Oops, thanks, I had missed it. I just pushed a commit for this.

**Comment by graelo** [comment]
Thank you!
